dist/gcc-compatible/Hacl_Streaming_SHA2.c

```diff
@@ -28,7 +28,7 @@
 #include "internal/Hacl_SHA2_Generic.h"
 #include "internal/Hacl_Krmllib.h"
 
-static inline void sha256_init(uint32_t *hash)
+void Hacl_SHA2_Scalar32_sha256_init(uint32_t *hash)
```

Comment:
I guess `static inline` can be restored by removing a `Hacl_Hash_SHA2.h` file and fixing calls in `EverCrypt_`?

Reply:
that would still remove the `static inline` because there would be a cross-compilation unit call from EverCrypt_HMAC.c to Hacl_Hash_SHA2.c, meaning the init function would not be static inline

if there are performance concerns, we can always do

```
let sha256_init_public st = sha256_init st
```

and then EverCrypt_HMAC.c calls `sha256_init_public` (marked `extern`), while `sha256_init` remains `static inline`

what do you think?

Reply:
Should we fix `mk_compute` in this PR, or can it be done in a further PR?

Reply:
if this fix is quick/simple, we can do it in this PR

Reply:
The fix is not simple. We do not have a way to say "I need any *instance* of the streaming functor that happens to exactly produce the hash functions". We can say "I want the MD streaming functor (non-agile, non-multiplexing)", or "I want the Evercrypt streaming functor (agile & multiplexing)", but we don't have a generic quantification that covers both.

I'm sure it's feasible, but seeing that I got good perf with just bundling stuff together, I'd rather not write super-complicated type signatures that will be hard to maintain in any case. Hope that's ok with you? Let me know!